tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlobStoreBackedReadHandleImpl.java

```diff
@@ -160,7 +170,14 @@ public CompletableFuture<LedgerEntries> readAsync(long firstEntry, long lastEntr
                         // the read method, we should allow to seek to the right position and the entry id should
                         // never over to the last entry again.
                         if (!seeked) {
-                            inputStream.seek(index.getIndexEntryForEntry(nextExpectedId).getDataOffset());
+                            Long knownOffset = entryOffsets.get(nextExpectedId);
```

Comment:
Maybe we can a function to package this logic and all  the `index.getIndexEntryForEntry(nextExpectedId).getDataOffset()` call this function.

Other seek places also need to check the index cache first.

Reply:
sounds good, in my testing it looked that only this point needs the cache, btw it is better to use it everywhere